optuna/integration/allennlp/_train.py

```diff
@@ -0,0 +1,406 @@
+"""
+This file is ported from AllenNLP original implementation. (Under Apache 2.0 license)
+https://github.com/allenai/allennlp/blob/main/allennlp/commands/train.py
+"""
+
+import logging
+import os
+from os import PathLike
+from typing import Any
+from typing import cast
+from typing import List
+from typing import Optional
+from typing import Union
+
+from allennlp.commands.train import TrainModel
+from allennlp.common import logging as common_logging
+from allennlp.common import Params
+from allennlp.common import util as common_util
+from allennlp.common.checks import check_for_gpu
+from allennlp.common.checks import ConfigurationError
+from allennlp.common.plugins import import_plugins
+from allennlp.data import Vocabulary
+from allennlp.models.archival import archive_model
+from allennlp.models.archival import CONFIG_NAME
+from allennlp.models.archival import verify_include_in_archive
+from allennlp.models.model import _DEFAULT_WEIGHTS
+from allennlp.models.model import Model
+from allennlp.training import util as training_util
+import torch
+import torch.distributed as dist
+import torch.multiprocessing as mp
+
+from optuna import Trial
+from optuna import TrialPruned
+from optuna._imports import try_import
+from optuna.storages import _CachedStorage
+from optuna.storages import InMemoryStorage
+
+
+with try_import() as _imports:
+    from allennlp.common.meta import Meta
+    from allennlp.common.meta import META_NAME
+
+if not _imports.is_successful():
+    Meta = None  # type: ignore  # NOQA
+    META_NAME = None  # type: ignore  # NOQA
+
+
+logger = logging.getLogger(__name__)
+
+
+def train_model_with_optuna(
+    params: Params,
+    serialization_dir: Union[str, PathLike],
+    recover: bool = False,
+    force: bool = False,
+    node_rank: int = 0,
+    include_package: List[str] = None,
+    dry_run: bool = False,
+    file_friendly_logging: bool = False,
+    trial: Trial = None,
+) -> Optional[Model]:
+    """Extended AllenNLP training utility.
+
+    # Parameters
+
+    params : `Params`
+        A parameter object specifying an AllenNLP Experiment.
```

Comment:
How about adapting the docstring to Optuna's standard format? For example, 
```suggestion
    Args:
        params:
            A parameter object specifying an AllenNLP Experiment.
```

Reply:
Thank you. You suggestion and b76a997 update docstrings.